livekit-agents/livekit/agents/voice_assistant/assistant.py

```diff
@@ -481,13 +480,15 @@ def _on_final_transcript(self, text: str) -> None:
 
         # this speech may not be validated, so we create a copy
         # of our context to add the new user message
-        chat_ctx = copy.deepcopy(self._chat_ctx)
```

Comment:
opted for shallow copy because deepcopy has issues with images (i.e. rtc.VideoFrame isn't picklable and we don't want to copy those anyways)

Reply:
Ah, but we do want a copy, otherwise the code isn't right and all the final transcription will stack up as user message in the ChatContext. 

I think we can implement a internal copy fnc for ChatContext?

Reply:
We need a deep copy?

Reply:
There is also a deepcopy inside _answer_from_text, I think you missed it

Reply:
Is this to prevent a footgun for users or are we mutating ChatMessages in our implementation?

Reply:
We're mutating inside our impl

Reply:
Added a copy. Couldn't find any other copies that I missed...